### ISSUE TITLE
Firefox support

### ISSUE THREAD

**Issue by myfonj** [miragecraft/drykit#3]
Just reminder / headsup that at this moment the demo page (and your personal homepage) renders suboptimally in Firefox (tried in 114 and 115 Developer edition) in "wide" (1157 px) width:

https://www.miragecraft.com/projects/razorgrid.html :
![Firefox window showing demo page with strangely overflowing boxes](https://github.com/miragecraft/razorgrid/assets/1761395/cdba5d5c-bff1-47a9-9b5f-6573075e1202)

https://www.miragecraft.com/ : 
![two tightly packed narrow text columns, first overlapping on the other at few places](https://github.com/miragecraft/razorgrid/assets/1761395/b8a783f9-9b7f-4bfe-ab99-9cc6f5177222)

Both pages link https://www.miragecraft.com/css/razorgrid.css that reads `RazorGrid v3.0.1`.

**Comment by myfonj** [comment]
No, I am not using it, just randomly bumped into your site (you know through your csswg-drafts issue) and saw it broken in my daily driver Fx. (It is OK in narrower viewport and in Chrome.)

Seeing something working in Chrome and broken in Firefox usually makes me dive into it and extract what's wrong and then search / create bugreports, but ATM have very little time to do that and. So if you manage to test it and fix it in future version it would be neat if you could be able to tell what was the culprit. But no pressure. This issue was meant just as a gentle nod towards testing in multiple engines, and for the case someone stumbles upon this issue in the near future.

**Comment by miragecraft** [comment]
Ahhh got it, thanks for caring =)

I assure you I tested it in all three mainline browsers - Chrome, Firefox and Safari. It definitely was working fine in Firefox before, they must have broken something in one of the later updates. I'll take a closer look this evening.

Won't be the first time I encountered a browser bug haha...

**Comment by miragecraft** [comment]
After taking a brief look, I'm pretty sure it has to do with the way I relied on dividing by zero behaviour in CSS calc().

Dividing by zero used to be invalid, but the spec later changed it so that it becomes positive or negative infinity. While dividing zero by zero turns into NaN, which "infects" the entire calc() statement into NaN which finally resolves to 0.

In the (old) version of RazorGrid I'm using on the current site I relied on `0/0` to make max-width invalid when  `--span: 0`. But it looks like Firefox has implemented the updated spec on `0/0` behaviour to force max-width to 0, thereby breaking things.

I'll fix it soon.
